components/director/internal/repo/list_pageable.go

```diff
@@ -22,6 +22,7 @@ type PageableQuerier interface {
 // PageableQuerierGlobal is an interface for listing with paging of global entities.
 type PageableQuerierGlobal interface {
 	ListGlobal(ctx context.Context, pageSize int, cursor string, orderByColumn string, dest Collection, additionalConditions ...Condition) (*pagination.Page, int, error)
+	ListWithQueryGlobal(ctx context.Context, query string, args []interface{}, orderByColumn string, pageSize int, cursor string, dest Collection) (*pagination.Page, int, error)
```

Comment:
I don't think this is needed at all. You can just introduce `LIKE` condition and use the standard querier.
This way we will have a few added benefits:
- No SQL injection mistakes can happen as the logic is encapsulated and abstracted with conditions as with the other entities.
- No hardcoded SQL in the tenant repo.